README.md

```diff
@@ -1019,6 +1019,255 @@ Voilà! Since we updated `index.html`, you'll need to restart the dev server. Go
+let fetchFeed = (~username, ~page) => {
+  setState(state =>
+    {
+      ...state,
+      step:
+        switch (state.step) {
+        | Loaded(r)
+        | Loading(Some(r)) => Loading(Some(r))
+        | Loading(None)
+        | Idle
+        | InvalidUsername => Loading(None)
+        },
+    }
+  );
+  module P = Js.Promise;
+  Fetch.fetch(
+    "https://gh-feed.vercel.app/api?user="
+    ++ Username.toString(username)
+    ++ "&page="
+    ++ string_of_int(page),
+  )
+  |> P.then_(response => {
+        let status = Fetch.Response.status(response);
+        if (status === 200) {
+          /* If status is OK, proceed to parse the response */
+          response
+          |> Fetch.Response.text
+          |> P.then_(text =>
+              {
+                let data =
+                  try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                  | Json.Decode.DecodeError(msg) =>
+                    Js.Console.error(msg);
+                    Error("Failed to decode: " ++ msg);
+                  };
+
+                switch (data) {
+                | Error(_) =>
+                  setState(state => {...state, step: Loaded(data)})
+                | Ok(data) =>
+                  setState(state => {
+                    let updatedFeed =
+                      switch (state.step) {
+                      | Loaded(Ok(feed))
+                      | Loading(Some(Ok(feed))) =>
+                        mergeFeeds(feed, data)
+                      | _ => data
+                      };
+                    {
+                      ...state,
+                      step: Loaded(Ok(updatedFeed)),
+                      currentPage: page + 1,
+                    };
+                  })
+                };
+              }
+              |> P.resolve
+            );
+        } else {
+          /* Handle non-200 status */
+          ...
+        };
+      })
+  |> ignore;
+};
+```
+
+#### 11.5: Modify render code to include scroll sentinel and handling new state
+
+We’ll add a `sentinelRef` to track the last element of the list, which will
+trigger loading more data when it comes into view.
+
+In the render code, create a new React ref to track the last element of the
+list:
+
+```reason
+    let sentinelRef = React.useRef(Js.Nullable.null);
+```
+
+Also, add the element itself at the end of the `switch`:
+
+```reason
+  {switch (state.step) {
+    | InvalidUsername => <div> {React.string("Invalid username")} </div>
+    | Idle =>
+      <div>
+        {React.string(
+          "Press the \"Enter\" key to confirm the username selection.",
+        )}
+      </div>
+    | Loading(None | Some(Error(_))) =>
+      <div> {React.string("Loading...")} </div>
+    | Loading(Some(Ok(feed))) =>
+      <> {renderFeed(feed)} <div> {React.string("Loading...")} </div> </>
+    | Loaded(Error(msg)) => <div> {React.string(msg)} </div>
+    | Loaded(Ok(feed)) => renderFeed(feed)
+    }}
+  <div ref={ReactDOM.Ref.domRef(sentinelRef)} />
```

Comment:
Starting to think we should've just written this in Docusaurus or VitePress so we can get line highlighting...

Reply:
I meant the ability that Docusaurus and VitePress give you to specifically highlight certain lines to indicate that they're of interest or that they changed. Like in these code listings: https://react-book.melange.re/promo-component/#react-usereducer

Reply:
Ah yeah, that'd be nice...